File: spmercantil/src/main/java/github/faustofjunqueira/spmercantil/application/adapters/controller/request/FilterMarketRequest.java
package github.faustofjunqueira.spmercantil.application.adapters.controller.request;

import lombok.Data;

/**
 * Informações a serem filtradas sobre as feiras
 */
@Data
public class FilterMarketRequest {

    /**
     * Nome do Distrito Municipal
     */
    private String district;

    /**
     * Região conforme divisão do Município em 5 áreas
     */
    private String region5;

    /**
     * Denominação da feira livre atribuída pela Supervisão de Abastecimento
     */
    private String name;

    /**
     * Bairro de localização da feira livre
     */
    private String neighborhood;

}

File: SpMercantil/Application/Controller/Market/Dto/Request/FilterMarketRequest.cs
#nullable enable
using System.ComponentModel.DataAnnotations;
using Microsoft.AspNetCore.Mvc;

namespace Application.Controller.Market.Dto.Request
{
    /// <summary>
    ///     Objeto de filtro de feira(market)
    /// </summary>
    public class FilterMarketRequest
    {
        /// <summary>
        ///     Nome do Distrito Municipal
        /// </summary>
        [FromQuery(Name = "district")]
        [MaxLength(18)]
        public string? District { get; set; }

        /// <summary>
        ///     Região conforme divisão do Município em 5 áreas
        /// </summary>
        [FromQuery(Name = "region5")]
        [MaxLength(6)]
        public string? Region5 { get; set; }

        /// <summary>
        ///     Denominação da feira livre atribuída pela Supervisão de Abastecimento
        /// </summary>
        [FromQuery(Name = "name")]
        [MaxLength(30)]
        public string? Name { get; set; }

        /// <summary>
        ///     Bairro de localização da feira livre
        /// </summary>
        [FromQuery(Name = "neighborhood")]
        [MaxLength(20)]
        public string? Neighborhood { get; set; }

        /// <summary>
        ///     Pagina a ser acessada. default=0
        /// </summary>
        [FromQuery(Name = "page")]
        [Range(0, 1000000)]
        public int Page { get; set; } = 0;

        /// <summary>
        ///     Quantidade de registros na pagina default=25.
        /// </summary>
        [FromQuery(Name = "size")]
        [Range(0, 25)]
        public int Size { get; set; } = 25;
    }
}